pkg/e2e/fake_metrics_server_test.go

```diff
@@ -0,0 +1,315 @@
+		}},
+	}}
+	createTimeSeriesRequest := &monitoringpb.CreateTimeSeriesRequest{
+		Name:       "PROJECT/1234",
+		TimeSeries: timeSeries,
+	}
+	fms.CreateTimeSeries(context.TODO(), createTimeSeriesRequest)
+
+	// these are the subtests
+	tests := []*createTimeSeriesTest{
+		{testName: "TestNil"},
+		{
+			testName: "TestExceedMaxTimeSeriesPerRequest",
+			createTimeSeriesRequest: &monitoringpb.CreateTimeSeriesRequest{
+				Name:       "PROJECT/1234",
+				TimeSeries: []*monitoringpb.TimeSeries{{}, {}},
+			},
+		},
+		{
+			testName: "TestNoTimeSeriesToAdd",
+			createTimeSeriesRequest: &monitoringpb.CreateTimeSeriesRequest{
+				Name: "PROJECT/1234",
+			},
+		},
+		{
+			testName: "TestNoPointInTimeSeriesToAdd",
+			createTimeSeriesRequest: &monitoringpb.CreateTimeSeriesRequest{
+				Name: "PROJECT/1234",
+				TimeSeries: []*monitoringpb.TimeSeries{{
+					Resource: &monitoredrespb.MonitoredResource{
+						Type: "prometheus_target",
+						Labels: map[string]string{
+							"project_id": "example-project",
+							"location":   "europe",
+							"cluster":    "foo-cluster",
+							"namespace":  "",
+							"job":        "job1",
+							"instance":   "instance1",
+						},
+					},
+					Metric: &metricpb.Metric{
+						Type:   "prometheus.googleapis.com/metric1/gauge",
+						Labels: map[string]string{"k1": "v1"},
+					},
+					MetricKind: metricpb.MetricDescriptor_GAUGE,
+					ValueType:  metricpb.MetricDescriptor_DOUBLE,
+				}},
+			},
+		},
+		{
+			testName: "TestAddPointInPast",
+			createTimeSeriesRequest: &monitoringpb.CreateTimeSeriesRequest{
+				Name: "PROJECT/1234",
+				TimeSeries: []*monitoringpb.TimeSeries{{
+					Resource: &monitoredrespb.MonitoredResource{
+						Type: "prometheus_target",
+						Labels: map[string]string{
+							"project_id": "example-project",
+							"location":   "europe",
+							"cluster":    "foo-cluster",
+							"namespace":  "",
+							"job":        "job1",
+							"instance":   "instance1",
+						},
+					},
+					Metric: &metricpb.Metric{
+						Type:   "prometheus.googleapis.com/metric1/gauge",
+						Labels: map[string]string{"k1": "v1"},
+					},
+					MetricKind: metricpb.MetricDescriptor_GAUGE,
+					ValueType:  metricpb.MetricDescriptor_DOUBLE,
+					Points: []*monitoringpb.Point{{
+						Interval: &monitoringpb.TimeInterval{
+							StartTime: &timestamppb.Timestamp{Seconds: 1},
+							EndTime:   &timestamppb.Timestamp{Seconds: 2},
+						},
+						Value: &monitoringpb.TypedValue{
+							Value: &monitoringpb.TypedValue_DoubleValue{DoubleValue: 0.6},
+						},
+					}},
+				}},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			response, err := fms.CreateTimeSeries(context.TODO(), test.createTimeSeriesRequest)
+			if err == nil && response != nil {
+				t.Errorf("expected an error for %q", test.testName)
+			}
+		})
+	}
+}
+
+func TestCreateTimeSeries(t *testing.T) {
+	fms := NewFakeMetricServer(200)
+	projectName := "PROJECT/1234"
```

Comment:
keep the style consistent, either in previous test, use a project name instead of hardcoded string, or in this test, use the hardcoded string per test case.